browser/ui/views/tabs/brave_tab.cc

```diff
@@ -356,3 +367,21 @@ void BraveTab::AddLayerToBelowThis() {
 
   ReorderChildLayers(layer()->parent());
 }
+
+void BraveTab::UpdateBorder() {
```

Comment:
How about overriding UpdateIconVisibility() and setting extra_padding_before_content_ and extra_alert_indicator_padding_ to true? I feel like that would be more compliant with how upstream works.

Reply:
Actually, I tried that first but it broke layout in some cases like below.
To avoid it, maybe we need to patch more to `Tab::Layout()`.
But increasing border's left inset works well w/o these regressions.

alert icon is clipped(left) compared with right.
<img width="63" alt="Screenshot 2023-06-06 at 9 26 00 AM" src="https://github.com/brave/brave-core/assets/6786187/d4e7d680-fee1-4259-9627-26c5b0d1fa73"><img width="69" alt="Screenshot 2023-06-06 at 9 25 49 AM" src="https://github.com/brave/brave-core/assets/6786187/71458032-773d-4ecc-bb70-6b8f56cc4c5f">

Icon & alert indicator are too close with x button
<img width="78" alt="Screenshot 2023-06-06 at 9 24 34 AM" src="https://github.com/brave/brave-core/assets/6786187/57843bdc-94f4-436c-a1cf-f58af5d422fa">
<img width="63" alt="Screenshot 2023-06-06 at 9 25 05 AM" src="https://github.com/brave/brave-core/assets/6786187/46483061-efbb-4af7-b3e9-5b347ecdfaa2">

Reply:
Fair enough! thanks. Then optionally, I'd like to suggest this routines be moved to GetInsets() instead of setting new border. Setting new border could cause many ops like layout and painting, overriding GetInsets() could be better.

Reply:
Which `GetInsets()` in `Tab` class?
I think we need to update border based on the result of `UpdateIconVisibility()`.
Fortunately, `UpdateIconVisibility()` is only called at the start of `Layout()` and `UpdateBorder()` is called by `UpdateIconVisibility()`.
So, I think changing border would not cause additional layout?

Reply:
Ah, do you mean overriding `View::GetInsets()`? Curious how overriding `View::GetInsets()` achieves same goal?

Reply:
Ah, got it. like below? checking.
```
gfx::Insets BraveTab::GetInsets() const {
  int extra_left_padding = 0;

  // Add extra padding if upstream tab doesn't have it.
  if (!extra_padding_before_content_) {
    extra_left_padding = kExtraLeftPadding;
  }

  auto insets = Tab::GetInsets();
  insets.set_left(insets.left() + extra_left_padding);
  return insets;
}
```

Reply:
Ok, updated. Using GetInsets() seems more cleaner. Thanks!